File: dotnet/aws/Mcma.Aws.DynamoDb/DynamoDbTableProvider.cs
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Mcma.Data;

namespace Mcma.Aws.DynamoDb
{
    public class DynamoDbTableProvider : IDocumentDatabaseTableProvider
    {
        public DynamoDbTableProvider(DynamoDbTableOptions options = null, IAmazonDynamoDB dynamoDb = null)
        {
            Options = options ?? new DynamoDbTableOptions();
            DynamoDb = dynamoDb ?? new AmazonDynamoDBClient();
        }
        
        private DynamoDbTableOptions Options { get; }
        
        private IAmazonDynamoDB DynamoDb { get; }

        public async Task<IDocumentDatabaseTable> GetAsync(string tableName)
            => new DynamoDbTable(DynamoDb, await DynamoDb.GetTableDescriptionAsync(tableName), Options);
    }
}

File: nodejs/aws/dynamodb/lib/dynamo-db-table-provider.ts
import { DynamoDB } from "aws-sdk";
import { DocumentDatabaseTableProvider, DocumentDatabaseTable } from "@mcma/data";

import { DynamoDbTable } from "./dynamo-db-table";
import { DynamoDbTableOptions } from "./dynamo-db-table-options";
import { getTableDescription } from "./dynamo-db-table-description";

export class DynamoDbTableProvider implements DocumentDatabaseTableProvider {
    
    constructor(
        private readonly options: DynamoDbTableOptions = {},
        private readonly dynamoDb = new DynamoDB()) {
    }

    async get(tableName: string): Promise<DocumentDatabaseTable> {
        return new DynamoDbTable(this.dynamoDb, await getTableDescription(this.dynamoDb, tableName), this.options);
    }
}
